src/Analyzers/CSharp/Analyzers/UsePatternMatching/CSharpAsAndNullCheckDiagnosticAnalyzer.cs

```diff
@@ -104,26 +101,20 @@ private void SyntaxNodeAction(SyntaxNodeAnalysisContext syntaxContext)
             return;
         }
 
-        var localStatement = declarator.Parent?.Parent;
-        var enclosingBlock = localStatement?.Parent;
-        if (localStatement == null ||
-            enclosingBlock == null)
-        {
+        if (declarator is not { Parent.Parent: LocalDeclarationStatementSyntax localStatement })
             return;
-        }
 
-        // Bail out if the potential diagnostic location is outside the analysis span.
-        if (!syntaxContext.ShouldAnalyzeSpan(localStatement.Span))
+        // Don't convert if the as is part of a local declaration with a using keyword
+        // eg using var x = y as MyObject;
+        if (localStatement is LocalDeclarationStatementSyntax localDecl && localDecl.UsingKeyword != default)
             return;
 
-        // Don't convert if the as is part of a using statement
-        // eg using (var x = y as MyObject) { }
-        if (localStatement is UsingStatementSyntax)
```

Comment:
This check for old school using statements is no longer needed because the analyzer is restricted to declarators in a local declaration statement, right?

Reply:
correct. :)